sed/core/processor.py

```diff
@@ -299,9 +299,18 @@ def load(
             )
             self._dataframe = dataframe
             self._files = self.loader.files
+        elif kwds["runs"] is not None:
```

Comment:
I am not quite sure I follow. The processor class has no argument runs so the only way I saw it possible is with keyword arguments. 
Does adding runs as required entry for interface to files and folders make sense because runs, beamtimeID and daq already allows us to figure out the folder and file structure.

Reply:
A "runID" can in various contexts be a valid descriptor allowing a system to identify a given experimental dataset. Thus, I saw it as a meaningfull addition to providing a folder and a set of files. What I mean is to extend the whole interface in a consistent manner to allow for this.

Reply:
@rettigl I'd suggest if you can look at it. I have added the runs identifier everywhere and tried to change processor, but it breaks a lot of tests as they require files and folders. I tried to change it but not entirely sure. Otherwise we can discuss over a call.

Reply:
> On the other hand, why are we repeating an explicit call for each of these cases, when just adding files, folder and eventually runNumber to kwrds and calling a single `self.reader.load_dataframe(**kwrds)` would do the trick in much less lines?
> 
> I would suggest:
> 
> ```python
> if dataframe is not None:
>     self._dataframe = dataframe
> else:
>     dataframe, metadata = self.loader.read_dataframe(**kwds)
> ```
> 
> and leave to `read_dataframe` the error catching job if no valid method pointing at the data is used.

The reason for having all this code in processors is to have the option for injecting the transparent mirroring to a local storage by the mirrorUtil. For doing this, we need to resolve the file location and replace the corresponding path. All this functionality works with string paths, thus I would like to keep it that way. This is also more the natural way users would enter a path, I guess. What a reader does internally, I don't care...

Reply:
I'd say the copytool belongs more to the loader than to the processor. Then the processor can be simplified. Or perhaps I am missing something.

Regarding paths, I get your point. However, working with pathlib paths results in more readable and robust code. For users, they can still give a string and we apply what Steinn suggested

> I would suggest to make everything Path objects, and adding if isnstance(x,str): x = Path(x) everywhere.

Reply:
The idea of the tool is to be transparent to the loader, and work in the background, thus I implemented it into the processor. In mpes, we used it even standalone outside our classes. I think both issues are thinks to discuss with pros and cons, so I suggest you can open issues for them. They certainly don't belong into this PR.